mods/persistence/modules/hydroponics/trays.dm

```diff
@@ -0,0 +1,35 @@
+/obj/machinery/portable_atmospherics/hydroponics/before_save()
+	. = ..()
+	if(round(temp_chem_holder.reagents.total_volume) > 0)
+		CUSTOM_SV("temp_chem_holder", temp_chem_holder)
+	SAVE_SEED_OR_SEEDNAME(seed)
+
+	//Save time dependent stuff
+	var/saved_last_cycle = max(lastcycle - world.time, 0)
+	CUSTOM_SV("saved_last_cycle", saved_last_cycle)
+	var/saved_last_produce = max(lastproduce - world.time, 0)
+	CUSTOM_SV("saved_last_produce", saved_last_produce)
+
+/obj/machinery/portable_atmospherics/hydroponics/after_deserialize()
+	. = ..()
+	var/obj/saved_chemholder = LOAD_CUSTOM_SV("temp_chem_holder")
+	if(saved_chemholder)
+		temp_chem_holder = saved_chemholder
+	LOAD_SAVED_SEED_OR_SEEDNAME(seed)
+
+	lastcycle = LOAD_CUSTOM_SV("saved_last_cycle")
```

Comment:
Should these have world.time added to them on deserialization?

Reply:
Good catch!